### PR TITLE
ARROW-18291: [Release][Docs] Update how to release

### PR DESCRIPTION
**Pull request by kou**

### CODE REVIEW

**Comment by kou** [comment]
@raulcd @assignUser You may be interested in this.

File: docs/source/developers/release.rst [hunk 1]
```diff
@@ -216,123 +216,365 @@ Be sure to go through on the following checklist:
...
 #. Upload wheels/sdist to pypi
 #. Publish Maven artifacts
 #. Update R packages
 #. Update vcpkg port
+#. Update Conan recipe
 #. Bump versions
 #. Update tags for Go modules
 #. Update docs
+#. Announce the new release
 #. Remove old artifacts
```
Comment (raulcd) [inline docs/source/developers/release.rst:238]: Should we add the update of the Arrow version on the cookbooks as a post-release task too? The main problem is that is requires the conda packages to be available at the moment and they might take a while. I am monitoring for them to be available for 10.0.0 to update the cookbooks version at the moment.

File: docs/source/developers/release.rst [hunk 2]
```diff
@@ -216,123 +216,365 @@ Be sure to go through on the following checklist:
...
-    A PMC member must upload the binary release artifacts to Artifactory:
+   Open a pull request to Homebrew:
 
-    .. code-block::
-    
-        # dev/release/post-02-binary.sh 0.1.0 0
-        dev/release/post-02-binary.sh <version> <rc number>
+   .. code-block:: Bash
+
+      ## You need to run this on macOS that Homebrew is installed.
```
Comment (raulcd) [inline docs/source/developers/release.rst:343]: Minor nit, this should also work with brewlinux, that's how I did the 9.0.0 one

File: docs/source/developers/release.rst [hunk 3]
```diff
@@ -216,123 +216,365 @@ Be sure to go through on the following checklist:
...
+
+      # dev/release/post-10-go.sh 10.0.0
+      dev/release/post-10-go.sh X.Y.Z
+
+.. dropdown:: Update docs
+   :animate: fade-in-slide-down
+   :class-title: sd-fs-5
+   :class-container: sd-shadow-md
+
+   The API documentations are generated in release process. We just need to upload the generated API documentations:
```
Comment (raulcd) [inline docs/source/developers/release.rst:546]:
```suggestion
   The API documentations are generated in the release process. We just need to upload the generated API documentations:
```

**Review by raulcd — COMMENTED** [review]
Thanks @kou , some minor nits

File: docs/source/developers/release.rst (hunk 3 above)
Comment (jorisvandenbossche) [reply docs/source/developers/release.rst:546]: And you can also remove "API", since this is for all docs, not just API docs

**Comment by raulcd** [comment]
Changes can be found here from the pr-docs job: http://crossbow.voltrondata.com/pr_docs/14612/developers/release.html

File: docs/source/developers/release.rst [hunk 4]
```diff
@@ -216,123 +216,365 @@ Be sure to go through on the following checklist:
...
+      dev/release/post-01-upload.sh <version> <rc>
 
-.. dropdown:: Updating the Arrow website
+.. dropdown:: Upload binary release artifacts to Artifactory
    :animate: fade-in-slide-down
    :class-title: sd-fs-5
    :class-container: sd-shadow-md
 
-    Fork the `arrow-site repository <https://github.com/apache/arrow-site>`_ and clone it next to the arrow repository.
+   A PMC member must upload the binary release artifacts to Artifactory:
```
Comment (assignUser) [inline docs/source/developers/release.rst:302]: I thought committers also have access to the artifactory, not entirely sure though, might need to request the access, couldn't find anything in the infra wiki. :shrug:

**Review by assignUser — APPROVED** [review]
LGTM

File: docs/source/developers/release.rst [hunk 5]
```diff
@@ -216,123 +216,365 @@ Be sure to go through on the following checklist:
 #. Start the new version on JIRA on the ARROW project
 #. Start the new version on JIRA for the related CPP PARQUET version
 #. Merge changes on release branch to maintenance branch for patch releases
+#. Add the new release to the Apache Reporter System
 #. Upload source
 #. Upload binaries
 #. Update website
```
[the remark this answers is not in the document]
Comment (kou) [reply docs/source/developers/release.rst:222]: Good catch!
I've added a place holder for it. Could you fill it in a follow-up pull request?

File: docs/source/developers/release.rst (hunk 1 above)
Comment (kou) [reply docs/source/developers/release.rst:240]: Wow, I didn't know that!
 I've added a place holder for it. Could you fill it in a follow-up pull request?

File: docs/source/developers/release.rst (hunk 4 above)
Comment (kou) [reply docs/source/developers/release.rst]: Right. I've fixed.

File: docs/source/developers/release.rst (hunk 2 above)
Comment (kou) [reply docs/source/developers/release.rst]: Really!? I didn't know that!

I've added "or Linux".

File: docs/source/developers/release.rst (hunk 3 above)
Comment (kou) [reply docs/source/developers/release.rst]: OK. Fixed them.

**Pull request merged**